File: backend/Calory/Calory.Domain/HealthGoal.cs
using System.ComponentModel.DataAnnotations;

namespace Calory.Domain
{
    public class HealthGoal
    {
        [Key]
        public Guid Id { get; set; }

        public Guid UserId { get; set; }

        public decimal DailyCalorieTarget { get; set; }

        public decimal ProteinTarget { get; set; }

        public decimal CarbTarget { get; set; }

        public decimal FatTarget { get; set; }

        public decimal WeightTarget { get; set; }

        public DateOnly StartDate { get; set; }

        public DateOnly? EndDate { get; set; } 

        public bool IsActive { get; set; }

        public DateTime CreatedAt { get; set; }
    }
}

File: frontend/src/app/interfaces/health-goal.ts
export interface HealthGoal {
  id: string;
  dailyCalorieTarget: number;
  proteinTarget: number;
  carbTarget: number;
  fatTarget: number;
  weightTarget: number;
  startDate: string;
  endDate: string | null;
  isActive: boolean;
  createdAt: string;
}

export interface HealthGoalRequest {
  dailyCalorieTarget: number;
  proteinTarget: number;
  carbTarget: number;
  fatTarget: number;
  weightTarget: number;
  startDate: string;
  endDate?: string | null;
}
